### PR TITLE
Fix /profile fields update

### LINKED ISSUE #603: DEFECT: profile edit leftover tasks
- [ ] profile edit for agency description is not updating immediately after save
- [ ] validations for empty strings missing for some fields in the edit modal -> please test and add validation rules

### PR DESCRIPTION
**Pull request by nguyenit67**
Fix profile fields not update immediately after save 
and add validation rules for updating account and agency fields

resolves #603

### CODE REVIEW

**Review by Enubia — APPROVED** [review]
LGTM!

@stacysealky please verify the phone regex as I'm not familiar with US phone numbers, merge if it's fine for you

File: src/middleware/validations.ts [hunk 1]
```diff
@@ -107,27 +107,58 @@ export default class Validations {
...
-			body('fName', 'First name is required!').exists(),
-			body('lName', 'Last name is required!').exists(),
+			body('fName', 'First name is required!').trim().isLength({ min: 1 }).isString(),
+			body('lName', 'Last name is required!').trim().isLength({ min: 1 }).isString(),
 		];
 	}
 
+	private static readonly AGENCY_PHONE_NUMBER_REGEX =
+		/^\(?([0-9]{3})\)?([ .-]?)([0-9]{3})\2([0-9]{4})/;
+
```
Comment (stacysealky) [inline src/middleware/validations.ts:122]: could you change it to 
^[(]?[2-9][0-9]{2}[)]?[-\s]?[0-9]{3}[-\s]?[0-9]{4}$

US phone numbers cannot start with 0 or 1, by the way.

**Review by stacysealky — CHANGES_REQUESTED** [review]
Great work! Just one change requested for the regex :) Thanks

File: src/middleware/validations.ts [hunk 2]
```diff
@@ -107,27 +107,58 @@ export default class Validations {
...
 		return [
-			body('fName', 'First name is required!').exists(),
-			body('lName', 'Last name is required!').exists(),
+			body('fName', 'First name is required!').trim().isLength({ min: 1 }).isString(),
+			body('lName', 'Last name is required!').trim().isLength({ min: 1 }).isString(),
 		];
 	}
 
+	private static readonly AGENCY_PHONE_NUMBER_REGEX =
+		/^\(?([0-9]{3})\)?([ .-]?)([0-9]{3})\2([0-9]{4})/;
```
Comment (nguyenit67) [inline src/middleware/validations.ts:121]: ofc, this is the new code change, please take a look and check if it's correct yet
```suggestion
	private static readonly AGENCY_PHONE_NUMBER_REGEX =
		/^[(]?[2-9][0-9]{2}[)]?[-\s]?[0-9]{3}[-\s]?[0-9]{4}$/;
```

File: src/middleware/validations.ts (hunk 1 above)
Comment (nguyenit67) [reply src/middleware/validations.ts:122]: of course, this is the new code change, please take a look and check if it's correct yet
```suggestion
	private static readonly AGENCY_PHONE_NUMBER_REGEX =
		/^[(]?[2-9][0-9]{2}[)]?[-\s]?[0-9]{3}[-\s]?[0-9]{4}$/;

```

Comment (nguyenit67) [inline src/middleware/validations.ts:122]: @stacysealky of course, this is the new code change, please take a look and check if it's correct yet
```suggestion
	private static readonly AGENCY_PHONE_NUMBER_REGEX =
		/^[(]?[2-9][0-9]{2}[)]?[-\s]?[0-9]{3}[-\s]?[0-9]{4}$/;

```

Comment (Enubia) [reply src/middleware/validations.ts:122]: I actually would update it a bit further

```js
/^(\([2-9][0-9]{2}\)[-\s]|[2-9][0-9]{2}[-\s])?[0-9]{3}[-\s]?[0-9]{4}$/
```

this matches the following strings

`123-1222`
`(234)-123-1222`
`234-123-1222`
`(234) 123-1222`
`234 123-1222`

your solution would also match `(223-123-1222` `223)-123-1222` which is not correct I think

File: src/middleware/validations.ts (hunk 2 above)
Comment (Enubia) [inline src/middleware/validations.ts:121]:
```suggestion
		/^(\([2-9][0-9]{2}\)[-\s]|[2-9][0-9]{2}[-\s])?[0-9]{3}[-\s]?[0-9]{4}$/;
```

**Review by Enubia — CHANGES_REQUESTED** [review]

File: src/middleware/validations.ts (hunk 2 above)
Comment (Enubia) [reply src/middleware/validations.ts:121]: @stacysealky @nguyenit67

I actually would update it a bit further

```js
/^(\([2-9][0-9]{2}\)[-\s]|[2-9][0-9]{2}[-\s])?[0-9]{3}[-\s]?[0-9]{4}$/
```

this matches the following strings

`123-1222`
`(234)-123-1222`
`234-123-1222`
`(234) 123-1222`
`234 123-1222`

your solution would also match `(223-123-1222` `223)-123-1222` which is not correct I think

Comment (stacysealky) [reply src/middleware/validations.ts:121]: Hmm but mine doesn't match (223-123-1222 223)-123-1222. Did you test it?

Comment (Enubia) [reply src/middleware/validations.ts:121]: it does

yours matches this `(223-123-1222` and this `223)-123-1222`

Comment (stacysealky) [reply src/middleware/validations.ts:121]: Yeah ok, let's just get rid of the brackets and unify the format to 333-333-3333 only
That way we can add a mask to the input UI

^[2-9][0-9]{2}[-][0-9]{3}[-][0-9]{4}$

Comment (stacysealky) [reply src/middleware/validations.ts:121]: Also I thought you meant that it matches "(223-123-1222 223)-123-1222"
Didn't realize they were two separate strings @Enubia

Comment (stacysealky) [reply src/middleware/validations.ts:121]: @Enubia @nguyenit67  Let's just go with what Marco wrote
Sorry for the confusion haha
/^(\([2-9][0-9]{2}\)[-\s]|[2-9][0-9]{2}[-\s])?[0-9]{3}[-\s]?[0-9]{4}$/

**Comment by Enubia** [comment]
@stacysealky updated the regex, not sure if @nguyenit67 is available right now

**Comment by nguyenit67** [comment]
Sr for the late reply, I just finish my last subject exam

**Comment by nguyenit67** [comment]
Sr for the late reply, I've just finished my last subject's exam. Do we need to support the international phone number format also? @Enubia @stacysealky

**Pull request merged**